Index: app/src/main/java/ir/maziardev/chatrapp/activitys/MediaActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/ir/maziardev/chatrapp/activitys/MediaActivity.java	(revision 6cbe705c6d898ef8a865a920a4b1881de073eff9)
+++ app/src/main/java/ir/maziardev/chatrapp/activitys/MediaActivity.java	(revision 0cb4c9a2881cc39fe84173eef21c040048090460)
@@ -3,11 +3,9 @@
 
 import android.content.Context;
 import android.os.Bundle;
-import android.util.Log;
 import android.view.View;
 import android.widget.LinearLayout;
 import android.widget.ProgressBar;
-import android.widget.Toast;
 
 import androidx.appcompat.app.AppCompatActivity;
 import androidx.recyclerview.widget.DefaultItemAnimator;
@@ -50,6 +48,20 @@
 
     @BindView(R.id.progress_tv)
     ProgressBar progress_tv;
+    @BindView(R.id.progress_radio)
+    ProgressBar progress_radio;
+    @BindView(R.id.progress_movie)
+    ProgressBar progress_movie;
+    @BindView(R.id.progress_music)
+    ProgressBar progress_music;
+    @BindView(R.id.progress_series)
+    ProgressBar progress_series;
+    @BindView(R.id.progress_carton)
+    ProgressBar progress_carton;
+    @BindView(R.id.progress_quran)
+    ProgressBar progress_quran;
+    @BindView(R.id.progress_nohe)
+    ProgressBar progress_nohe;
 
     @BindView(R.id.recycler_tv_media)
     RecyclerView recycler_tv;
@@ -96,6 +108,20 @@
 
         progress_tv.clearAnimation();
         progress_tv.setVisibility(View.GONE);
+        progress_radio.clearAnimation();
+        progress_radio.setVisibility(View.GONE);
+        progress_movie.clearAnimation();
+        progress_movie.setVisibility(View.GONE);
+        progress_series.clearAnimation();
+        progress_series.setVisibility(View.GONE);
+        progress_carton.clearAnimation();
+        progress_carton.setVisibility(View.GONE);
+        progress_music.clearAnimation();
+        progress_music.setVisibility(View.GONE);
+        progress_quran.clearAnimation();
+        progress_quran.setVisibility(View.GONE);
+        progress_nohe.clearAnimation();
+        progress_nohe.setVisibility(View.GONE);
 
         init();
 
@@ -147,14 +173,14 @@
         mAdapter_tv.notifyDataSetChanged();
 
 
-        LinearLayoutManager finalLayoutManager = layoutManager;
+        LinearLayoutManager finalLayoutManagertv = layoutManager;
         recycler_tv.setOnScrollListener(new RecyclerView.OnScrollListener() {
 
             @Override
             public void onScrolled(RecyclerView recyclerView, int dx, int dy) {
                 super.onScrolled(recyclerView, dx, dy);
-                int totalItem = finalLayoutManager.getItemCount();
-                int lastVisibleItem = finalLayoutManager.findLastVisibleItemPosition();
+                int totalItem = finalLayoutManagertv.getItemCount();
+                int lastVisibleItem = finalLayoutManagertv.findLastVisibleItemPosition();
 
                 if (mAdapter_tv.getMoreDataAvailable() && !mLoading && lastVisibleItem == totalItem - 1) {
                     mLoading = true;
@@ -171,7 +197,7 @@
         mAdapter_radio = new ListTvAdapter(this, menuList_radio);
         layoutManager = new LinearLayoutManager(this, LinearLayoutManager.HORIZONTAL, true);
         recycler_radio.setLayoutManager(layoutManager);
-        recycler_radio.setItemAnimator(new DefaultItemAnimator());
+        //recycler_radio.setItemAnimator(new DefaultItemAnimator());
         recycler_radio.setAdapter(mAdapter_radio);
 
         for (int i = 0; i < AppController.list_radio.size(); i++) {
@@ -185,6 +211,25 @@
         }
         mAdapter_radio.notifyDataSetChanged();
 
+        LinearLayoutManager finalLayoutManagerradio = layoutManager;
+        recycler_radio.setOnScrollListener(new RecyclerView.OnScrollListener() {
+
+            @Override
+            public void onScrolled(RecyclerView recyclerView, int dx, int dy) {
+                super.onScrolled(recyclerView, dx, dy);
+                int totalItem = finalLayoutManagerradio.getItemCount();
+                int lastVisibleItem = finalLayoutManagerradio.findLastVisibleItemPosition();
+                if (mAdapter_radio.getMoreDataAvailable() && !mLoading && lastVisibleItem == totalItem - 1) {
+                    mLoading = true;
+
+                    AppController.PAGE_RADIO += 1;
+                    loadMore(2, AppController.PAGE_RADIO);
+                }
+            }
+        });
+
+
+
 
         //__________________________________________movie
         mAdapter_movie = new ListTvAdapter(this, menuList_movie);
@@ -204,6 +249,23 @@
         }
         mAdapter_movie.notifyDataSetChanged();
 
+        LinearLayoutManager finalLayoutManagermovie = layoutManager;
+        recycler_movie.setOnScrollListener(new RecyclerView.OnScrollListener() {
+
+            @Override
+            public void onScrolled(RecyclerView recyclerView, int dx, int dy) {
+                super.onScrolled(recyclerView, dx, dy);
+                int totalItem = finalLayoutManagermovie.getItemCount();
+                int lastVisibleItem = finalLayoutManagermovie.findLastVisibleItemPosition();
+                if (mAdapter_movie.getMoreDataAvailable() && !mLoading && lastVisibleItem == totalItem - 1) {
+                    mLoading = true;
+
+                    AppController.PAGE_MOVIE += 1;
+                    loadMore(3, AppController.PAGE_MOVIE);
+                }
+            }
+        });
+
 
         //__________________________________________series
         mAdapter_series = new ListTvAdapter(this, menuList_series);
@@ -223,6 +285,23 @@
         }
         mAdapter_series.notifyDataSetChanged();
 
+        LinearLayoutManager finalLayoutManagerseries = layoutManager;
+        recycler_tv.setOnScrollListener(new RecyclerView.OnScrollListener() {
+
+            @Override
+            public void onScrolled(RecyclerView recyclerView, int dx, int dy) {
+                super.onScrolled(recyclerView, dx, dy);
+                int totalItem = finalLayoutManagerseries.getItemCount();
+                int lastVisibleItem = finalLayoutManagerseries.findLastVisibleItemPosition();
+                if (mAdapter_tv.getMoreDataAvailable() && !mLoading && lastVisibleItem == totalItem - 1) {
+                    mLoading = true;
+
+                    AppController.PAGE_SERIES += 1;
+                    loadMore(4, AppController.PAGE_SERIES);
+                }
+            }
+        });
+
 
         //__________________________________________carton
         mAdapter_carton = new ListTvAdapter(this, menuList_carton);
@@ -242,6 +321,24 @@
         }
         mAdapter_carton.notifyDataSetChanged();
 
+        LinearLayoutManager finalLayoutManagercarton = layoutManager;
+        recycler_tv.setOnScrollListener(new RecyclerView.OnScrollListener() {
+
+            @Override
+            public void onScrolled(RecyclerView recyclerView, int dx, int dy) {
+                super.onScrolled(recyclerView, dx, dy);
+                int totalItem = finalLayoutManagercarton.getItemCount();
+                int lastVisibleItem = finalLayoutManagercarton.findLastVisibleItemPosition();
+                if (mAdapter_tv.getMoreDataAvailable() && !mLoading && lastVisibleItem == totalItem - 1) {
+                    mLoading = true;
+
+                    AppController.PAGE_CARTON += 1;
+                    loadMore(5, AppController.PAGE_CARTON);
+                }
+            }
+        });
+
+
         //__________________________________________music
         mAdapter_music = new ListTvAdapter(this, menuList_music);
         layoutManager = new LinearLayoutManager(this, LinearLayoutManager.HORIZONTAL, true);
@@ -260,6 +357,23 @@
         }
         mAdapter_music.notifyDataSetChanged();
 
+        LinearLayoutManager finalLayoutManagermusic = layoutManager;
+        recycler_tv.setOnScrollListener(new RecyclerView.OnScrollListener() {
+
+            @Override
+            public void onScrolled(RecyclerView recyclerView, int dx, int dy) {
+                super.onScrolled(recyclerView, dx, dy);
+                int totalItem = finalLayoutManagermusic.getItemCount();
+                int lastVisibleItem = finalLayoutManagermusic.findLastVisibleItemPosition();
+                if (mAdapter_tv.getMoreDataAvailable() && !mLoading && lastVisibleItem == totalItem - 1) {
+                    mLoading = true;
+
+                    AppController.PAGE_MUSIC += 1;
+                    loadMore(6, AppController.PAGE_MUSIC);
+                }
+            }
+        });
+
 
         //__________________________________________quran
         mAdapter_quran = new ListTvAdapter(this, menuList_quran);
@@ -279,6 +393,23 @@
         }
         mAdapter_quran.notifyDataSetChanged();
 
+        LinearLayoutManager finalLayoutManagerquran = layoutManager;
+        recycler_tv.setOnScrollListener(new RecyclerView.OnScrollListener() {
+
+            @Override
+            public void onScrolled(RecyclerView recyclerView, int dx, int dy) {
+                super.onScrolled(recyclerView, dx, dy);
+                int totalItem = finalLayoutManagerquran.getItemCount();
+                int lastVisibleItem = finalLayoutManagerquran.findLastVisibleItemPosition();
+                if (mAdapter_tv.getMoreDataAvailable() && !mLoading && lastVisibleItem == totalItem - 1) {
+                    mLoading = true;
+
+                    AppController.PAGE_QURAN += 1;
+                    loadMore(7, AppController.PAGE_QURAN);
+                }
+            }
+        });
+
 
         //__________________________________________nohe
         mAdapter_nohe = new ListTvAdapter(this, menuList_nohe);
@@ -297,6 +428,23 @@
             menuList_nohe.add(mainlist);
         }
         mAdapter_nohe.notifyDataSetChanged();
+
+        LinearLayoutManager finalLayoutManagernohe = layoutManager;
+        recycler_tv.setOnScrollListener(new RecyclerView.OnScrollListener() {
+
+            @Override
+            public void onScrolled(RecyclerView recyclerView, int dx, int dy) {
+                super.onScrolled(recyclerView, dx, dy);
+                int totalItem = finalLayoutManagernohe.getItemCount();
+                int lastVisibleItem = finalLayoutManagernohe.findLastVisibleItemPosition();
+                if (mAdapter_tv.getMoreDataAvailable() && !mLoading && lastVisibleItem == totalItem - 1) {
+                    mLoading = true;
+
+                    AppController.PAGE_NOHE += 1;
+                    loadMore(8, AppController.PAGE_NOHE);
+                }
+            }
+        });
     }
 
 
@@ -308,25 +456,38 @@
                 loadMoreTV(index);
                 break;
             case 2:
-
+                progress_radio.clearAnimation();
+                progress_radio.setVisibility(View.VISIBLE);
+                loadMoreRadio(index);
                 break;
             case 3:
-
+                progress_movie.clearAnimation();
+                progress_movie.setVisibility(View.VISIBLE);
+                loadMoreTV(index);
                 break;
             case 4:
-
+                progress_series.clearAnimation();
+                progress_series.setVisibility(View.VISIBLE);
                 break;
             case 5:
-
+                progress_carton.clearAnimation();
+                progress_carton.setVisibility(View.VISIBLE);
+                loadMoreTV(index);
                 break;
             case 6:
-
+                progress_music.clearAnimation();
+                progress_music.setVisibility(View.VISIBLE);
+                loadMoreTV(index);
                 break;
             case 7:
-
+                progress_quran.clearAnimation();
+                progress_quran.setVisibility(View.VISIBLE);
+                loadMoreTV(index);
                 break;
             case 8:
-
+                progress_nohe.clearAnimation();
+                progress_nohe.setVisibility(View.VISIBLE);
+                loadMoreTV(index);
                 break;
         }
     }
@@ -359,13 +520,6 @@
                                 media.setFlag(true);
                                 media.setTintcolor(true);
 
-                                /*dbHelperTvMedia.insertData(
-                                        media.getId_category(),
-                                        media.getTitle(),
-                                        media.getImg(),
-                                        media.getUrl(),
-                                        true
-                                );*/
 
                                 AppController.list_tel.add(media);
                                 Mainlist mainlist = new Mainlist();
@@ -399,6 +553,66 @@
         AppController.getInstance().addToRequestQueue(req, "loadMoreTv");
     }
 
+    private void loadMoreRadio(int page){
+        StringRequest req = new StringRequest(Request.Method.GET, AppController.API_MEDIA_URL + "radio/"  + "1/" + AppController.API_BASE_LIMIT + "/" +  AppController.APP_TOKEN,
+                new Response.Listener<String>() {
+                    @Override
+                    public void onResponse(String response) {
+
+                        try {
+                            JSONObject obj = new JSONObject(response);
+                            JSONArray radioArray = obj.getJSONArray("radio");
+
+                            if(radioArray.length()== 0 || radioArray.length()>= 10){
+                                mAdapter_radio.setMoreDataAvailable(false);
+                                AppController.PAGE_RADIO -= 1;
+                            }
+
+                            for (int i = 0; i < radioArray.length(); i++) {
+                                JSONObject Object = radioArray.getJSONObject(i);
+
+                                Lists media = new Lists();
+                                media.setTitle(Object.getString("title"));
+                                media.setImg(Object.getString("img"));
+                                media.setUrl(Object.getString("url"));
+
+                                media.setSite(true);
+                                media.setFlag(true);
+                                media.setRadio(true);
+
+                                AppController.list_radio.add(media);
+                                Mainlist mainlist = new Mainlist();
+                                mainlist.setTitle(media.getTitle());
+                                mainlist.setUrl(media.getUrl());
+                                mainlist.setImg(media.getImg());
+                                mainlist.setMainType(MainType.RADIO);
+                                menuList_radio.add(mainlist);
+                            }
+
+
+                            mLoading = false;
+                            mAdapter_radio.notifyDataSetChanged();
+
+                            progress_radio.clearAnimation();
+                            progress_radio.setVisibility(View.GONE);
+
+
+                        } catch (JSONException e) {
+                            e.printStackTrace();
+                        } catch (Exception e) {
+                            e.printStackTrace();
+                        }
+                    }
+                }, new Response.ErrorListener() {
+            @Override
+            public void onErrorResponse(VolleyError error) {
+                VolleyLog.d(TAG, "Error: " + error.getMessage());
+            }
+        });
+        req.setShouldCache(false);
+        AppController.getInstance().addToRequestQueue(req, "loadMoreRadio");
+    }
+
 
     @Override
     protected void attachBaseContext(Context newBase) {
Index: app/src/main/java/ir/maziardev/chatrapp/activitys/SplashActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/ir/maziardev/chatrapp/activitys/SplashActivity.java	(revision 6cbe705c6d898ef8a865a920a4b1881de073eff9)
+++ app/src/main/java/ir/maziardev/chatrapp/activitys/SplashActivity.java	(revision 0cb4c9a2881cc39fe84173eef21c040048090460)
@@ -127,16 +127,16 @@
 
         if (save.load(AppController.SAVE_STORAGE_PERMISION, "0").equals("0")) showStorageDialog();
 
-        initDb();
+        //initDb();
         if (isNetworkConnected()) {
-            initClearDb();
+            //initClearDb();
             scheduler.scheduleAtFixedRate(new CustomTask(), 1000, 1000, TimeUnit.MILLISECONDS);
         } else {
             Toast.makeText(this, "اتصال به اینترنت برقرار نیست!", Toast.LENGTH_LONG).show();
-            getAllDataFromDB();
+           /* getAllDataFromDB();
             startActivity(new Intent(SplashActivity.this, MainActivity.class));
             scheduler.shutdown();
-            finish();
+            finish();*/
         }
 
     }
@@ -499,13 +499,13 @@
                                 category.setId(object.getString("id"));
                                 category.setTitle(object.getString("title"));
 
-                                dbCategory.insertData(
+                                /*dbCategory.insertData(
                                         category.getId(),
                                         category.getTitle()
-                                );
+                                );*/
                                 AppController.arrayList_category.add(category);
                             }
-                            dbCategory.close();
+                            //dbCategory.close();
 
                             initAll();
 
@@ -548,16 +548,16 @@
                                 library.setImg(object.getString("img"));
                                 library.setUrl(object.getString("url"));
 
-                                dbAudiobookLib.insertData(
+                                /*dbAudiobookLib.insertData(
                                         library.getId_category(),
                                         library.getTitle(),
                                         library.getImg(),
                                         library.getUrl(),
                                         library.isSite()
-                                );
+                                );*/
                                 AppController.list_li_audio.add(library);
                             }
-                            dbAudiobookLib.close();
+                            //dbAudiobookLib.close();
 
                             for (int i = 0; i < quranJsonList.length(); i++) {
                                 JSONObject object = quranJsonList.getJSONObject(i);
@@ -567,16 +567,16 @@
                                 library.setImg(object.getString("img"));
                                 library.setUrl(object.getString("url"));
 
-                                dbQuranLib.insertData(
+                                /*dbQuranLib.insertData(
                                         library.getTitle(),
                                         library.getImg(),
                                         library.getUrl(),
                                         library.isSite()
-                                );
+                                );*/
 
                                 AppController.list_li_quran.add(library);
                             }
-                            dbQuranLib.close();
+                            //dbQuranLib.close();
 
                             for (int i = 0; i < mafatihJsonList.length(); i++) {
                                 JSONObject object = mafatihJsonList.getJSONObject(i);
@@ -586,15 +586,15 @@
                                 library.setImg(object.getString("img"));
                                 library.setUrl(object.getString("url"));
 
-                                dbMafatihLib.insertData(
+                                /*dbMafatihLib.insertData(
                                         library.getTitle(),
                                         library.getImg(),
                                         library.getUrl(),
                                         library.isSite()
-                                );
+                                );*/
                                 AppController.list_li_mafatih.add(library);
                             }
-                            dbMafatihLib.close();
+                            //dbMafatihLib.close();
 
                             for (int i = 0; i < nahjJsonList.length(); i++) {
                                 JSONObject object = nahjJsonList.getJSONObject(i);
@@ -604,15 +604,15 @@
                                 library.setImg(object.getString("img"));
                                 library.setUrl(object.getString("url"));
 
-                                dbNahjLib.insertData(
+                                /*dbNahjLib.insertData(
                                         library.getTitle(),
                                         library.getImg(),
                                         library.getUrl(),
                                         library.isSite()
-                                );
+                                );*/
                                 AppController.list_li_nahj.add(library);
                             }
-                            dbNahjLib.close();
+                            //dbNahjLib.close();
 
                             for (int i = 0; i < resaleJsonList.length(); i++) {
                                 JSONObject object = resaleJsonList.getJSONObject(i);
@@ -622,15 +622,15 @@
                                 library.setImg(object.getString("img"));
                                 library.setUrl(object.getString("url"));
 
-                                dbResaleLib.insertData(
+                                /*dbResaleLib.insertData(
                                         library.getTitle(),
                                         library.getImg(),
                                         library.getUrl(),
                                         library.isSite()
-                                );
+                                );*/
                                 AppController.list_li_resale.add(library);
                             }
-                            dbResaleLib.close();
+                            //dbResaleLib.close();
 
                             downloaditem++;
                         } catch (JSONException e) {
@@ -666,16 +666,16 @@
                                 service.setImg(object.getString("img"));
                                 service.setUrl(object.getString("url"));
 
-                                dbService.insertData(
+                                /*dbService.insertData(
                                         service.getId_category(),
                                         service.getTitle(),
                                         service.getImg(),
                                         service.getUrl(),
                                         service.isSite()
-                                );
+                                );*/
                                 AppController.list_services.add(service);
                             }
-                            dbService.close();
+                            //dbService.close();
 
                             downloaditem++;
 
@@ -714,16 +714,16 @@
                                 game.setId_category(object.getString("id_category"));
                                 game.setUrl(object.getString("url"));
 
-                                dbGames.insertData(
+                                /*dbGames.insertData(
                                         game.getId_category(),
                                         game.getTitle(),
                                         game.getImg(),
                                         game.getUrl(),
                                         game.isSite()
-                                );
+                                );*/
                                 AppController.list_games.add(game);
                             }
-                            dbGames.close();
+                            //dbGames.close();
 
                             downloaditem++;
                         } catch (JSONException e) {
@@ -757,13 +757,13 @@
                                 magazin.setTitle(object.getString("title"));
                                 magazin.setUrl(object.getString("url"));
 
-                                dbMagazin.insertData(
+                                /*dbMagazin.insertData(
                                         magazin.getTitle(),
                                         magazin.getUrl()
-                                );
+                                );*/
                                 AppController.arrayList_magazin.add(magazin);
                             }
-                            dbMagazin.close();
+                            //dbMagazin.close();
 
                             downloaditem++;
                         } catch (JSONException e) {
@@ -798,15 +798,15 @@
                                 slider.setImg(sliderObject.getString("img"));
                                 slider.setAction(sliderObject.getString("action"));
 
-                                dbSlider.insertData(
+                                /*dbSlider.insertData(
                                         slider.getTitle(),
                                         slider.getDescription(),
                                         slider.getImg(),
                                         slider.getAction()
-                                );
+                                );*/
                                 AppController.arrayList_slider.add(slider);
                             }
-                            dbSlider.close();
+                            //dbSlider.close();
 
                             downloaditem++;
                         } catch (JSONException e) {
@@ -869,16 +869,16 @@
                                         break;
                                 }
 
-                                dbHelperChannel.insertData(
-                                        channel.getTitle(),
-                                        channel.getImg(),
-                                        channel.getMusic(),
-                                        channel.getMovie(),
-                                        channel.getSurvey(),
-                                        channel.getLocation(),
-                                        channel.getPhone(),
-                                        channelObject.getString("type")
-                                );
+//                                dbHelperChannel.insertData(
+//                                        channel.getTitle(),
+//                                        channel.getImg(),
+//                                        channel.getMusic(),
+//                                        channel.getMovie(),
+//                                        channel.getSurvey(),
+//                                        channel.getLocation(),
+//                                        channel.getPhone(),
+//                                        channelObject.getString("type")
+//                                );
                                 AppController.arrayList_channel.add(channel);
                             }
                             downloaditem++;
@@ -917,15 +917,15 @@
                                 weatherCity.setLat(object.getString("lat"));
                                 weatherCity.setLng(object.getString("lng"));
 
-                                dbHelperWeather.insertData(
-                                        weatherCity.getProvince(),
-                                        weatherCity.getStationEn(),
-                                        weatherCity.getStationFa(),
-                                        weatherCity.getStationNumber(),
-                                        weatherCity.getiCAO(),
-                                        weatherCity.getLat(),
-                                        weatherCity.getLng()
-                                );
+//                                dbHelperWeather.insertData(
+//                                        weatherCity.getProvince(),
+//                                        weatherCity.getStationEn(),
+//                                        weatherCity.getStationFa(),
+//                                        weatherCity.getStationNumber(),
+//                                        weatherCity.getiCAO(),
+//                                        weatherCity.getLat(),
+//                                        weatherCity.getLng()
+//                                );
                                 AppController.arrayList_weather_city.add(weatherCity);
                             }
 
@@ -966,13 +966,13 @@
                                 media.setFlag(true);
                                 media.setTintcolor(true);
 
-                                dbHelperTvMedia.insertData(
+                                /*dbHelperTvMedia.insertData(
                                         media.getId_category(),
                                         media.getTitle(),
                                         media.getImg(),
                                         media.getUrl(),
                                         media.isSite()
-                                );
+                                );*/
                                 AppController.list_tel.add(media);
                             }
                             downloaditem++;
@@ -1014,13 +1014,13 @@
                                 media.setFlag(true);
                                 media.setRadio(true);
 
-                                dbHelperRadioMedia.insertData(
+                                /*dbHelperRadioMedia.insertData(
                                         media.getId_category(),
                                         media.getTitle(),
                                         media.getImg(),
                                         media.getUrl(),
                                         media.isSite()
-                                );
+                                );*/
                                 AppController.list_radio.add(media);
                             }
                             downloaditem++;
@@ -1064,14 +1064,14 @@
                                 Categoryy category = AppController.arrayList_category.get(Integer.parseInt(Object.getString("id_category")));
                                 media.setCategory(category.getTitle());
 
-                                dbHelperMusicMedia.insertData(
+                                /*dbHelperMusicMedia.insertData(
                                         media.getCategory(),
                                         media.getTitle(),
                                         media.getImg(),
                                         media.getUrl(),
                                         media.getActor(),
                                         media.isSite()
-                                );
+                                );*/
                                 AppController.list_music.add(media);
                             }
                             downloaditem++;
@@ -1113,14 +1113,14 @@
                                 Categoryy category = AppController.arrayList_category.get(Integer.parseInt(Object.getString("id_category")));
                                 media.setCategory(category.getTitle());
 
-                                dbHelperMovieMedia.insertData(
+                                /*dbHelperMovieMedia.insertData(
                                         media.getCategory(),
                                         media.getTitle(),
                                         media.getImg(),
                                         media.getUrl(),
                                         "",
                                         false
-                                );
+                                );*/
                                 AppController.list_movie.add(media);
                             }
                             downloaditem++;
@@ -1163,14 +1163,14 @@
                                 media.setCategory(category.getTitle());
                                 media.setSeriesname(Object.getString("seriesname"));
 
-                                dbHelperSeriesMedia.insertData(
-                                        media.getCategory(),
-                                        media.getTitle(),
-                                        media.getImg(),
-                                        media.getUrl(),
-                                        media.getSeriesname(),
-                                        false
-                                );
+//                                dbHelperSeriesMedia.insertData(
+//                                        media.getCategory(),
+//                                        media.getTitle(),
+//                                        media.getImg(),
+//                                        media.getUrl(),
+//                                        media.getSeriesname(),
+//                                        false
+//                                );
                                 AppController.list_series.add(media);
                             }
                             downloaditem++;
@@ -1213,14 +1213,14 @@
                                 media.setCategory(category.getTitle());
                                 media.setSeriesname(Object.getString("seriesname"));
 
-                                dbHelperCartonMedia.insertData(
+                                /*dbHelperCartonMedia.insertData(
                                         media.getCategory(),
                                         media.getTitle(),
                                         media.getImg(),
                                         media.getUrl(),
                                         media.getSeriesname(),
                                         false
-                                );
+                                );*/
                                 AppController.list_carton.add(media);
                             }
                             downloaditem++;
@@ -1263,14 +1263,14 @@
                                 media.setSite(true);
                                 media.setMusic(true);
 
-                                dbHelperQuranMedia.insertData(
+                                /*dbHelperQuranMedia.insertData(
                                         media.getCategory(),
                                         media.getTitle(),
                                         media.getImg(),
                                         media.getUrl(),
                                         media.getActor(),
                                         false
-                                );
+                                );*/
                                 AppController.list_quran.add(media);
                             }
                             downloaditem++;
@@ -1313,14 +1313,14 @@
                                 media.setSite(true);
                                 media.setMusic(true);
 
-                                dbHelperNoheMedia.insertData(
+                                /*dbHelperNoheMedia.insertData(
                                         media.getCategory(),
                                         media.getTitle(),
                                         media.getImg(),
                                         media.getUrl(),
                                         media.getActor(),
                                         false
-                                );
+                                );*/
                                 AppController.list_nohe.add(media);
                             }
                             downloaditem++;
@@ -1365,12 +1365,12 @@
                                 media.setUrl(Object.getString("url"));
                                 media.setAction(Object.getString("action"));
 
-                                dbHelperChef.insertData(
+                                /*dbHelperChef.insertData(
                                         media.getTitle(),
                                         "",
                                         media.getImg(),
                                         media.getAction()
-                                );
+                                );*/
                                 AppController.list_sa_chef.add(media);
                             }
 
@@ -1383,12 +1383,12 @@
                                 media.setUrl(Object.getString("url"));
                                 media.setAction(Object.getString("action"));
 
-                                dbHelperEmploy.insertData(
+                                /*dbHelperEmploy.insertData(
                                         media.getTitle(),
                                         "",
                                         media.getImg(),
                                         media.getAction()
-                                );
+                                );*/
                                 AppController.list_sa_employ.add(media);
                             }
 
@@ -1401,12 +1401,12 @@
                                 media.setUrl(Object.getString("url"));
                                 media.setAction(Object.getString("action"));
 
-                                dbHelperSport.insertData(
+                                /*dbHelperSport.insertData(
                                         media.getTitle(),
                                         "",
                                         media.getImg(),
                                         media.getAction()
-                                );
+                                );*/
                                 AppController.list_sa_sport.add(media);
                             }
 
@@ -1419,12 +1419,12 @@
                                 media.setUrl(Object.getString("url"));
                                 media.setAction(Object.getString("action"));
 
-                                dbHelperFood.insertData(
+                                /*dbHelperFood.insertData(
                                         media.getTitle(),
                                         "",
                                         media.getImg(),
                                         media.getAction()
-                                );
+                                );*/
                                 AppController.list_sa_food.add(media);
                             }
 
@@ -1437,12 +1437,12 @@
                                 media.setUrl(Object.getString("url"));
                                 media.setAction(Object.getString("action"));
 
-                                dbHelperPlant.insertData(
+                                /*dbHelperPlant.insertData(
                                         media.getTitle(),
                                         "",
                                         media.getImg(),
                                         media.getAction()
-                                );
+                                );*/
                                 AppController.list_sa_plant.add(media);
                             }
 
@@ -1455,12 +1455,12 @@
                                 media.setUrl(Object.getString("url"));
                                 media.setAction(Object.getString("action"));
 
-                                dbHelperPray.insertData(
+                                /*dbHelperPray.insertData(
                                         media.getTitle(),
                                         "",
                                         media.getImg(),
                                         media.getAction()
-                                );
+                                );*/
                                 AppController.list_sa_pray.add(media);
                             }
 
@@ -1473,12 +1473,12 @@
                                 media.setUrl(Object.getString("url"));
                                 media.setAction(Object.getString("action"));
 
-                                dbHelperPsychology.insertData(
+                                /*dbHelperPsychology.insertData(
                                         media.getTitle(),
                                         "",
                                         media.getImg(),
                                         media.getAction()
-                                );
+                                );*/
                                 AppController.list_sa_psychology.add(media);
                             }
 
@@ -1535,13 +1535,13 @@
                                 media.setImg(Object.getString("img"));
                                 media.setUrl(Object.getString("url"));
 
-                                dbHelperIsna.insertData(
+                                /*dbHelperIsna.insertData(
                                         "1",
                                         media.getTitle(),
                                         media.getImg(),
                                         media.getUrl(),
                                         true
-                                );
+                                );*/
                                 AppController.list_ne_isna.add(media);
                             }
 
@@ -1553,13 +1553,13 @@
                                 media.setImg(Object.getString("img"));
                                 media.setUrl(Object.getString("url"));
 
-                                dbHelperJamjam.insertData(
+                                /*dbHelperJamjam.insertData(
                                         "1",
                                         media.getTitle(),
                                         media.getImg(),
                                         media.getUrl(),
                                         true
-                                );
+                                );*/
                                 AppController.list_ne_jamjam.add(media);
                             }
 
@@ -1571,13 +1571,13 @@
                                 media.setImg(Object.getString("img"));
                                 media.setUrl(Object.getString("url"));
 
-                                dbHelperTasnim.insertData(
+                                /*dbHelperTasnim.insertData(
                                         "1",
                                         media.getTitle(),
                                         media.getImg(),
                                         media.getUrl(),
                                         true
-                                );
+                                );*/
                                 AppController.list_ne_tasnim.add(media);
                             }
 
Index: app/src/main/java/ir/maziardev/chatrapp/network/AppController.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/ir/maziardev/chatrapp/network/AppController.java	(revision 6cbe705c6d898ef8a865a920a4b1881de073eff9)
+++ app/src/main/java/ir/maziardev/chatrapp/network/AppController.java	(revision 0cb4c9a2881cc39fe84173eef21c040048090460)
@@ -62,13 +62,13 @@
     public static final String WEATHER_CITY_URL = "http://mobileapp.irimo.ir/WeatherServer/GetStationList";
     public static final String WEATHER_BY_CITY_URL = "http://mobileapp.irimo.ir/WeatherServer/GetStationData/";
 
-    public static final String API_AZAN_URL = "http://chatr.anzalepoxy.ir/azan/index.htm";
-    public static final String API_EST_URL = "http://chatr.anzalepoxy.ir/estekhare/index.php";
-    public static final String API_SUPPORT_URL = "http://chatr.anzalepoxy.ir/chat/spachat.php";
+    public static final String API_AZAN_URL = "https://numberfa.ir/azan/index.htm";
+    public static final String API_EST_URL = "https://numberfa.ir/estekhare/index.php";
+    public static final String API_SUPPORT_URL = "https://numberfa.ir/chat/spachat.php";
 
     public static final String API_BASE_LIMIT = "7";
 
-    public static final String API_BASE_URL = "http://chatr.anzalepoxy.ir/chatr/app/index.php/api/";
+    public static final String API_BASE_URL = "https://numberfa.ir/chatr/app/index.php/api/";
     public static final String API_UPDATE_URL = API_BASE_URL + "update/";
     public static final String API_SLIDER_URL = API_BASE_URL + "slider/detail/";
     public static final String API_CHANNEL_URL = API_BASE_URL + "channel/detail/";
@@ -129,6 +129,13 @@
 
 
     public static int PAGE_TV = 1;
+    public static int PAGE_RADIO = 1;
+    public static int PAGE_MOVIE = 1;
+    public static int PAGE_SERIES = 1;
+    public static int PAGE_CARTON = 1;
+    public static int PAGE_MUSIC = 1;
+    public static int PAGE_QURAN = 1;
+    public static int PAGE_NOHE = 1;
 
 
 
Index: app/src/main/res/layout/activity_media.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/layout/activity_media.xml	(revision 6cbe705c6d898ef8a865a920a4b1881de073eff9)
+++ app/src/main/res/layout/activity_media.xml	(revision 0cb4c9a2881cc39fe84173eef21c040048090460)
@@ -78,12 +78,26 @@
                     android:text="پخش زنده رادیو"
                     android:textSize="15sp" />
 
-                <androidx.recyclerview.widget.RecyclerView
-                    android:id="@+id/recycler_radio_media"
-                    android:layout_width="match_parent"
-                    android:layout_height="wrap_content"
-                    android:layout_gravity="right" />
+                <LinearLayout
+                    android:layout_width="match_parent"
+                    android:layout_height="match_parent"
+                    android:orientation="horizontal">
+
+                    <ProgressBar
+                        android:id="@+id/progress_radio"
+                        android:layout_width="50dp"
+                        android:layout_height="match_parent"
+                        android:layout_gravity="center_horizontal"
+                        tools:visibility="gone" />
+
+                    <androidx.recyclerview.widget.RecyclerView
+                        android:id="@+id/recycler_radio_media"
+                        android:layout_width="match_parent"
+                        android:layout_height="wrap_content"
+                        android:layout_gravity="right" />
 
+                </LinearLayout>
+
             </LinearLayout>
 
             <LinearLayout
@@ -101,14 +115,28 @@
                     android:text="فیلم سینمایی"
                     android:textSize="15sp" />
 
-                <androidx.recyclerview.widget.RecyclerView
-                    android:id="@+id/recycler_movie_media"
-                    android:layout_width="match_parent"
-                    android:layout_height="wrap_content"
-                    android:layout_gravity="right" />
+                <LinearLayout
+                    android:layout_width="match_parent"
+                    android:layout_height="match_parent"
+                    android:orientation="horizontal">
+
+                    <ProgressBar
+                        android:id="@+id/progress_movie"
+                        android:layout_width="50dp"
+                        android:layout_height="match_parent"
+                        android:layout_gravity="center_horizontal"
+                        tools:visibility="gone" />
+
+                    <androidx.recyclerview.widget.RecyclerView
+                        android:id="@+id/recycler_movie_media"
+                        android:layout_width="match_parent"
+                        android:layout_height="wrap_content"
+                        android:layout_gravity="right" />
 
-            </LinearLayout>
+                </LinearLayout>
 
+            </LinearLayout>
+
             <LinearLayout
                 android:layout_width="match_parent"
                 android:layout_height="wrap_content"
@@ -124,14 +152,28 @@
                     android:text="سریال"
                     android:textSize="15sp" />
 
-                <androidx.recyclerview.widget.RecyclerView
-                    android:id="@+id/recycler_series_media"
-                    android:layout_width="match_parent"
-                    android:layout_height="wrap_content"
-                    android:layout_gravity="right" />
+                <LinearLayout
+                    android:layout_width="match_parent"
+                    android:layout_height="match_parent"
+                    android:orientation="horizontal">
+
+                    <ProgressBar
+                        android:id="@+id/progress_series"
+                        android:layout_width="50dp"
+                        android:layout_height="match_parent"
+                        android:layout_gravity="center_horizontal"
+                        tools:visibility="gone" />
+
+                    <androidx.recyclerview.widget.RecyclerView
+                        android:id="@+id/recycler_series_media"
+                        android:layout_width="match_parent"
+                        android:layout_height="wrap_content"
+                        android:layout_gravity="right" />
 
-            </LinearLayout>
+                </LinearLayout>
 
+            </LinearLayout>
+
             <LinearLayout
                 android:layout_width="match_parent"
                 android:layout_height="wrap_content"
@@ -147,14 +189,28 @@
                     android:text="کارتون"
                     android:textSize="15sp" />
 
-                <androidx.recyclerview.widget.RecyclerView
-                    android:id="@+id/recycler_carton_media"
-                    android:layout_width="match_parent"
-                    android:layout_height="wrap_content"
-                    android:layout_gravity="right" />
+                <LinearLayout
+                    android:layout_width="match_parent"
+                    android:layout_height="match_parent"
+                    android:orientation="horizontal">
+
+                    <ProgressBar
+                        android:id="@+id/progress_carton"
+                        android:layout_width="50dp"
+                        android:layout_height="match_parent"
+                        android:layout_gravity="center_horizontal"
+                        tools:visibility="gone" />
+
+                    <androidx.recyclerview.widget.RecyclerView
+                        android:id="@+id/recycler_carton_media"
+                        android:layout_width="match_parent"
+                        android:layout_height="wrap_content"
+                        android:layout_gravity="right" />
 
-            </LinearLayout>
+                </LinearLayout>
 
+            </LinearLayout>
+
             <LinearLayout
                 android:layout_width="match_parent"
                 android:layout_height="wrap_content"
@@ -170,14 +226,28 @@
                     android:text="موسیقی"
                     android:textSize="15sp" />
 
-                <androidx.recyclerview.widget.RecyclerView
-                    android:id="@+id/recycler_music_media"
-                    android:layout_width="match_parent"
-                    android:layout_height="wrap_content"
-                    android:layout_gravity="right" />
+                <LinearLayout
+                    android:layout_width="match_parent"
+                    android:layout_height="match_parent"
+                    android:orientation="horizontal">
+
+                    <ProgressBar
+                        android:id="@+id/progress_music"
+                        android:layout_width="50dp"
+                        android:layout_height="match_parent"
+                        android:layout_gravity="center_horizontal"
+                        tools:visibility="gone" />
+
+                    <androidx.recyclerview.widget.RecyclerView
+                        android:id="@+id/recycler_music_media"
+                        android:layout_width="match_parent"
+                        android:layout_height="wrap_content"
+                        android:layout_gravity="right" />
 
-            </LinearLayout>
+                </LinearLayout>
 
+            </LinearLayout>
+
             <LinearLayout
                 android:layout_width="match_parent"
                 android:layout_height="wrap_content"
@@ -193,14 +263,28 @@
                     android:text="قرآن"
                     android:textSize="15sp" />
 
-                <androidx.recyclerview.widget.RecyclerView
-                    android:id="@+id/recycler_quran_media"
-                    android:layout_width="match_parent"
-                    android:layout_height="wrap_content"
-                    android:layout_gravity="right" />
+                <LinearLayout
+                    android:layout_width="match_parent"
+                    android:layout_height="match_parent"
+                    android:orientation="horizontal">
+
+                    <ProgressBar
+                        android:id="@+id/progress_quran"
+                        android:layout_width="50dp"
+                        android:layout_height="match_parent"
+                        android:layout_gravity="center_horizontal"
+                        tools:visibility="gone" />
+
+                    <androidx.recyclerview.widget.RecyclerView
+                        android:id="@+id/recycler_quran_media"
+                        android:layout_width="match_parent"
+                        android:layout_height="wrap_content"
+                        android:layout_gravity="right" />
 
-            </LinearLayout>
+                </LinearLayout>
 
+            </LinearLayout>
+
             <LinearLayout
                 android:layout_width="match_parent"
                 android:layout_height="wrap_content"
@@ -216,12 +300,26 @@
                     android:text="نوحه"
                     android:textSize="15sp" />
 
-                <androidx.recyclerview.widget.RecyclerView
-                    android:id="@+id/recycler_nohe_media"
-                    android:layout_width="match_parent"
-                    android:layout_height="wrap_content"
-                    android:layout_gravity="right" />
+                <LinearLayout
+                    android:layout_width="match_parent"
+                    android:layout_height="match_parent"
+                    android:orientation="horizontal">
+
+                    <ProgressBar
+                        android:id="@+id/progress_nohe"
+                        android:layout_width="50dp"
+                        android:layout_height="match_parent"
+                        android:layout_gravity="center_horizontal"
+                        tools:visibility="gone" />
+
+                    <androidx.recyclerview.widget.RecyclerView
+                        android:id="@+id/recycler_nohe_media"
+                        android:layout_width="match_parent"
+                        android:layout_height="wrap_content"
+                        android:layout_gravity="right" />
 
+                </LinearLayout>
+
             </LinearLayout>
 
 
